### PR TITLE
feat: item(row) wise tax amount rounding

### PR DESCRIPTION
**Pull request by rtdany10**
Account Settings:
image

Before:

https://github.com/frappe/erpnext/assets/52111700/81e18cb4-1c10-4cc2-9d99-3f6cbcfa6e0d

After:

https://github.com/frappe/erpnext/assets/52111700/9c30126d-2a43-4854-a925-208b70d244b0

Some countries require tax to be shown on an item level in invoice. ref: https://github.com/frappe/erpnext/blob/bf6bf79ae3f475cb0e6d0793ea9781d24eb10c83/erpnext/regional/united_arab_emirates/utils.py#L33-L35

And the tax amounts are to be rounded 2 digits on item level itself and the total tax should match the summation of item-wise tax. This PR allows users to have item-level tax rounding if required.

`no-docs`

### CODE REVIEW

**Comment by rtdany10** [comment]
Linters failing on unrelated files

**Comment by rtdany10** [comment]
@deepeshgarg007 @ruthra-kumar

File: erpnext/accounts/doctype/accounts_settings/accounts_settings.json [hunk 1]
```diff
@@ -58,7 +59,6 @@
   "closing_settings_tab",
   "period_closing_settings_section",
   "acc_frozen_upto",
-  "ignore_account_closing_balance",
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/accounts_settings/accounts_settings.json]: @rtdany10 This got removed, please add back

Comment (rtdany10) [reply erpnext/accounts/doctype/accounts_settings/accounts_settings.json]: Sorry, have fixed

**Comment by rtdany10** [comment]
@deepeshgarg007

**Comment by rtdany10** [comment]
@deepeshgarg007

**Comment by rtdany10** [comment]
@deepeshgarg007

**Pull request merged**